lib/client-assets.php

```diff
@@ -693,3 +693,19 @@ function gutenberg_extend_settings_custom_units( $settings ) {
 	register_block_pattern( 'core/its-time', gutenberg_load_block_pattern( 'its-time' ) );
 	register_block_pattern( 'core/testimonials', gutenberg_load_block_pattern( 'testimonials' ) );
 }
+
+/**
+ * Extends block editor settings to include UI labels to be shown for elements.
+ *
+ * Currently experimental.
+ *
+ * @param  array $settings Default editor settings.
+ * @return array           Modified editor settings.
+ */
+function gutenberg_extend_settings_ui_labels( $settings ) {
+	$settings['__experimentalLabels'] = array(
```

Comment:
ℹ️ I had a brief thought of introducing new filter here - just for labels. Something like `apply_filters( 'block_editor_label_setting', $labels )` but then I thought it will be redundant. Vendor plugins can use `block_editor_settings` with lower priority to achieve the same.